stellargraph/connector/neo4j/graph.py

```diff
@@ -58,25 +66,94 @@ def nodes(self):
         result = self.graph_db.run(node_ids_query)
         return [row["node_id"] for row in result.data()]
 
-    def node_features(self, node_ids):
+    def cache_nodes(self, dtype="float32"):
+        nodes = self.nodes()
```

Comment:
This ends up doing two O(num nodes) queries, in here to get the list of nodes, and then in `_node_features_from_db`. In addition, the one in `_node_features_from_db` is "unnecessarily" having to use `UNWIND` to look up each node in order. I wonder if it's better to try to do a single one, somehow? For instance, `_node_features_from_db(None)` could include the node IDs too (e.g. return an `IndexedArray` instead of a NumPy array directly).